src/Metric/MetricFactory.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Baldinof\RoadRunnerBundle\Metric;
+
+use Baldinof\RoadRunnerBundle\Exception\UnknownRpcTransportException;
+use Spiral\Goridge\RPC;
+use Spiral\Goridge\SocketRelay;
+use Spiral\RoadRunner\Metrics;
+use Spiral\RoadRunner\MetricsInterface;
+
+class MetricFactory
+{
+    /**
+     * @var bool
+     */
+    private $metricsEnabled;
+
+    /**
+     * @var string
+     */
+    private $rrRpc;
+
+    /**
+     * @var string
+     */
+    private $kernelProjectDir;
+
+    public function __construct(string $rrRpc, string $kernelProjectDir, bool $metricsEnabled)
+    {
+        $this->rrRpc = $rrRpc;
+        $this->kernelProjectDir = $kernelProjectDir;
+        $this->metricsEnabled = $metricsEnabled;
+    }
+
+    public function getMetricService(): MetricsInterface
+    {
+        $nullMetrics = new NullMetrics();
+
+        if (!$this->metricsEnabled || !$this->rrRpc) {
+            return $nullMetrics;
+        }
+
+        $rpcDsn = parse_url($this->rrRpc);
+        if (!is_array($rpcDsn) || !array_key_exists('scheme', $rpcDsn)) {
+            return $nullMetrics;
```

Comment:
We can throw a `UnknownRpcTransportException` here, as the DSN should be parseable.

Reply:
Done